qiskit_machine_learning/primitives/kernels/quantum_kernel.py

```diff
@@ -0,0 +1,225 @@
+from qiskit.primitives import Sampler
+from qiskit.algorithms.fidelities import BaseFidelity, Fidelity
+
+from .base_kernel import BaseKernel
+
+
+class QuantumKernel(BaseKernel):
+    r"""
+    QuantumKernel
+
+    The general task of machine learning is to find and study patterns in data. For many
+    algorithms, the datapoints are better understood in a higher dimensional feature space,
+    through the use of a kernel function:
+
+    .. math::
+
+        K(x, y) = \langle f(x), f(y)\rangle.
+
+    Here K is the kernel function, x, y are n dimensional inputs. f is a map from n-dimension
+    to m-dimension space. :math:`\langle x, y \rangle` denotes the dot product.
+    Usually m is much larger than n.
+
+    The quantum kernel algorithm calculates a kernel matrix, given datapoints x and y and feature
+    map f, all of n dimension. This kernel matrix can then be used in classical machine learning
+    algorithms such as support vector classification, spectral clustering or ridge regression.
+
+    Here, the kernel function is defined as the overlap of two quantum states defined by a
+    parametrized quantum circuit (called feature map):
+
+    .. math::
+
+        K(x,y) = |\langle \phi(x) | \phi(y) \rangle|^2
+    """
+
+    def __init__(
+        self,
+        sampler: Sampler | None = None,
+        feature_map: QuantumCircuit | None = None,
+        fidelity: str | BaseFidelity = "zero_prob",
+        enforce_psd: bool = True,
+        evaluate_duplicates: str = "off_diagonal",
+    ) -> None:
+        super().__init__(enforce_psd=enforce_psd)
+
+        if feature_map is None:
+            feature_map = ZZFeatureMap(2)
+
+        self._num_features = feature_map.num_parameters
+        eval_duplicates = evaluate_duplicates.lower()
+        if eval_duplicates not in ("all", "off_diagonal", "none"):
+            raise ValueError(
+                f"Unsupported value passed as evaluate_duplicates: {evaluate_duplicates}"
+            )
+        self._evaluate_duplicates = eval_duplicates
+
+        if isinstance(fidelity, str):
+            if sampler is None:
+                raise ValueError(
+                    "If the fidelity is passed as a string, a sampler has to be provided"
+                    "(currently set to None)."
+                )
+            if fidelity == "zero_prob":
+                self._fidelity = Fidelity(
+                    sampler=sampler, left_circuit=feature_map, right_circuit=feature_map
+                )
+            else:
+                raise ValueError(
+                    f"{fidelity} is not a valid string for a fidelity. Currently supported: 'zero_prob'."
+                )
+        else:
+            if sampler is not None:
+                warnings.warn(
+                    "Passed both a sampler and a fidelity instance. If passing a fidelity instance"
+                    " for `fidelity`, the sampler will not be used.",
+                )
+            self._fidelity = fidelity
```

Comment:
Since it seems only a Sampler or Fidelity is allowed really despite being two parameters might it be worth considering to merge to a single parameter - Sampler | Fidelity - not sure what the name would be. Given a sampler it will always use the standard/default fidelity we create internally, if you want to use a different one then you pass a Fidelity instance containing the Sampler. This gets rid of str values too which is not great when it comes to IDE autocomplete